# - Change available size from 32, 40, 48, 64 to 24, 30, 36, 48
# - Remove command avalibility checks because `which` could not detect `nativeBuildInputs`
--- a/build.sh
+++ b/build.sh
@@ -1,66 +1,13 @@
 #! /usr/bin/env bash
 
-# from where the command was run
-ROOT=$(pwd)
-
-# check command avalibility
-has_command() {
-	command -v "$1" >/dev/null 2>&1
-}
-
-if [ ! "$(which xcursorgen 2> /dev/null)" ]; then
-	echo xorg-xcursorgen needs to be installed to generate the cursors.
-	if has_command zypper; then
-		sudo zypper install -y xcursorgen
-	elif has_command apt-get; then
-		sudo apt-get install -y xorg-xcursorgen || sudo apt-get install -y x11-apps
-	elif has_command dnf; then
-		sudo dnf install -y xcursorgen
-	elif has_command yum; then
-		sudo dnf install -y xcursorgen
-	elif has_command pacman; then
-		sudo pacman -S --noconfirm xorg-xcursorgen
-	fi
-fi
-
-if [ ! "$(which rsvg-convert 2> /dev/null)" ]; then
-	echo rsvg-convert needs to be installed to generate the cursors.
-	if has_command zypper; then
-		sudo zypper install -y rsvg-convert
-	elif has_command apt-get; then
-		sudo apt-get install -y librsvg2-bin
-	elif has_command dnf; then
-		sudo dnf install -y librsvg2 librsvg2-tools
-	elif has_command yum; then
-		sudo dnf install -y librsvg2 librsvg2-tools
-	elif has_command pacman; then
-		sudo pacman -S --noconfirm librsvg
-	fi
-fi
-
-if [ ! "$(which python3 2> /dev/null)" ]; then
-	echo python3 needs to be installed to generate svg cursors.
-	if has_command zypper; then
-		sudo zypper install -y python3
-	elif has_command apt-get; then
-		sudo apt-get install -y python3
-	elif has_command dnf; then
-		sudo dnf install -y python3
-	elif has_command yum; then
-		sudo dnf install -y python3
-	elif has_command pacman; then
-		sudo pacman -S --noconfirm python
-	fi
-fi
-
 function create {
 	cd "$SRC"
 	mkdir -p x1 x1_25 x1_5 x2
 	cd "$SRC"/$1
-	find . -name "*.svg" -exec sh -c 'rsvg-convert -w 32 -h 32 "$0" -o "../x1/$(basename "$0" .svg).png"' {} \;
-	find . -name "*.svg" -exec sh -c 'rsvg-convert -w 40 -h 40 "$0" -o "../x1_25/$(basename "$0" .svg).png"' {} \;
-	find . -name "*.svg" -exec sh -c 'rsvg-convert -w 48 -h 48 "$0" -o "../x1_5/$(basename "$0" .svg).png"' {} \;
-	find . -name "*.svg" -exec sh -c 'rsvg-convert -w 64 -h 64 "$0" -o "../x2/$(basename "$0" .svg).png"' {} \;
+	find . -name "*.svg" -exec sh -c 'rsvg-convert -w 24 -h 24 "$0" -o "../x1/$(basename "$0" .svg).png"' {} \;
+	find . -name "*.svg" -exec sh -c 'rsvg-convert -w 30 -h 30 "$0" -o "../x1_25/$(basename "$0" .svg).png"' {} \;
+	find . -name "*.svg" -exec sh -c 'rsvg-convert -w 36 -h 36 "$0" -o "../x1_5/$(basename "$0" .svg).png"' {} \;
+	find . -name "*.svg" -exec sh -c 'rsvg-convert -w 48 -h 48 "$0" -o "../x2/$(basename "$0" .svg).png"' {} \;
 
 	cd $SRC
 
@@ -113,19 +60,7 @@ function create {
 }
 
 # generate pixmaps from svg source
-SRC=$ROOT/src
+SRC=$(pwd)/src
 THEME="WhiteSur Cursors"
 
-function svg-cursors {
-	cd $ROOT
-	rm -rf ./svg-cursor/
-	rm -rf ./dist/cursors_scalable/
-	git clone https://github.com/jinliu/svg-cursor.git
-
-	echo -e "Generating SVG cursors...\\r"
-	./svg-cursor/build-svg-theme/build-svg-theme.py --output-dir=$ROOT/dist/cursors_scalable --svg-dir=$SRC/svg --config-dir=$SRC/config --alias-file=$SRC/cursorList --nominal-size=24 >/dev/null 2>&1
-	echo -e "Generating SVG cursors... DONE"
-}
-
 create svg
-svg-cursors
